test_runner/regress/test_image_layers.py

```diff
@@ -0,0 +1,47 @@
+import os
+import shutil
+from contextlib import closing
+from datetime import datetime
+from pathlib import Path
+from typing import List
+
+import pytest
+from fixtures.neon_fixtures import NeonEnv, wait_for_last_flush_lsn
+
+
+def test_image_layer_fail_before_finish(neon_simple_env: NeonEnv):
+    pageserver_http = neon_simple_env.pageserver.http_client()
+    pageserver_http.configure_failpoints(("image-layer-fail-before-finish", "return"))
+
+    image_files = []
+
+    with pytest.raises(Exception):
```

Comment:
What exactly fails here? I would not expect tenant and branch creation to fail in the test, only the compaction method.
So we might want to move this `with` to the smallest possible scope around the exception triggering method.